src/main/java/net/snowflake/client/core/arrow/fullvectorconverters/BigIntVectorConverter.java

```diff
@@ -0,0 +1,41 @@
+package net.snowflake.client.core.arrow.fullvectorconverters;
+
+import net.snowflake.client.core.DataConversionContext;
+import net.snowflake.client.core.SFBaseSession;
+import net.snowflake.client.core.SFException;
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.core.arrow.ArrowVectorConverter;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.BigIntVector;
+import org.apache.arrow.vector.ValueVector;
+
+@SnowflakeJdbcInternalApi
+public class BigIntVectorConverter extends SimpleArrowFullVectorConverter<BigIntVector> {
+
+  public BigIntVectorConverter(
+      RootAllocator allocator,
+      ValueVector vector,
+      DataConversionContext context,
+      SFBaseSession session,
+      int idx) {
+    super(allocator, vector, context, session, idx);
+  }
+
+  @Override
+  protected boolean matchingType() {
```

Comment:
nit: this helper method is a bit confusing to me, maybe it could be static and take ValueVector as an argument? something like `isVectorOfMatchingType(ValueVector vector)` to be more descriptive? But that's just a personal preference

Reply:
I don't have strong opinions one way or the other, I'll consider changing it

Reply:
Actually, it's not really possible doe to the need for it to be overriden - the convert() method depends on calling the matchingType() from subclass.